core/trino-main/src/main/java/io/trino/sql/planner/plan/TableWriterNode.java

```diff
@@ -55,7 +53,7 @@
     private final Symbol fragmentSymbol;
     private final List<Symbol> columns;
     private final List<String> columnNames;
-    private final Set<Symbol> notNullColumnSymbols;
+    private final List<Symbol> notNullColumnSymbols;
```

Comment:
Why the change here?
Does the order on this collection matter?

(seems that most of the changes are technical and related to this)

Reply:
So basically I had to make a choice out of 3 options:
1. Change ` map(node.getNotNullColumnSymbols())` in `SymbolMapper` line 406  in such a way that `node.getNotNullColumnSymbols()` is converted to List and then output of this map call is converted back to Set.
2. add another implementation of map function that would accept Set
3. do what I did, also note that other collections here are Lists too

Reply:
> add another implementation of map function that would accept Set

that would be fine if we could assume `SymbolMapper.map(Symbol)` function is _distinct_.


> 1. Change ` map(node.getNotNullColumnSymbols())` in `SymbolMapper` line 406  in such a way that `node.getNotNullColumnSymbols()` is converted to List and then output of this map call is converted back to Set.

I'd do probably do this.

> also note that other collections here are Lists too

the fields like
```
private final List<Symbol> columns;
private final List<String> columnNames;
```

generally look as describing some fields, so are index-correlated (perhaps)

our `notNullColumnSymbols` doesn't fall into this category, so it's better to be a Set to avoid people assuming some meaning that it doesn't convey.

Reply:
> 1. Change ` map(node.getNotNullColumnSymbols())` in `SymbolMapper` line 406  in such a way that `node.getNotNullColumnSymbols()` is converted to List and then output of this map call is converted back to Set.

Let's do this (a minimal change) so that we can focus on what the PR actually does and then think what to do.